style_guide/README.md

```diff
@@ -0,0 +1,200 @@
+
+
+git & GitHub Workflow
+---------------------
+
+The best method for contributing software to the `imap_processing` repository is
+a workflow that involves forking the repository, developing changes on "feature"
+branches, and opening pull requests through GitHub.
+
+The following diagram depicts this workflow (credit to
+[Atlassian](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow):
+
+<img src="https://wac-cdn.atlassian.com/dam/jcr:cc0b526e-adb7-4d45-874e-9bcea9898b4a/04%20Hotfix%20branches.svg?cdnVersion=1089" alt="git and GitHub Workflow" width="300" height="300">
+
+As such, all feature branches should be branched off of and merged back into the
+`dev` branch.
+
+### Contributing new features
+
+*Note: Steps (1) through (4) only to be done once.*
+
+1. Create a personal fork of the `imap_processing` repository by visiting the
+   repository main `IMAP-Science-Operations-Center` `imap_processing`
+   [repository](https://github.com/IMAP-Science-Operations-Center/imap_processing)
+   and clicking the `Fork` button near the top-right of the page.  Follow the
+   various steps to create a fork under your GitHub account.
+2. Make a local copy of your personal fork by cloning the repository, using the
+   URL found by clicking the green "clone" button.
+3. Ensure that the personal fork is pointing to the `upstream` `imap_processing`
+   repository:
+
+       git remote add upstream <url>
+
+4. Retrieve the `upstream` `dev` branch:
+
+       git fetch upstream
+       git checkout -b dev upstream/dev
+
+6. Create a feature branch off of the `dev` branch to develop changes on.
+   Branch names should be short but descriptive (e.g.
+   `update-codice-unit-tests`) and not too generic (e.g. `bug-fix`, `updates`).
+   Consistent use of hyphens is encouraged.
+
+        git checkout -b <branchname>
+
+7. Make changes to the branch using the nominal `git add`/`git commit` cycle:
+
+        git add <new or changed files you want to commit>
+        git commit -m 'Explaination of the changes'
+
+8. Push the feature branch to your personal fork's GitHub repository:
+
+        git push origin <branchname>
+
+9. On the `IMAP-Science-Operations-Center` `imap_processing`
+   [repository](https://github.com/IMAP-Science-Operations-Center/imap_processing)
+   create a new pull request.  Click on the "compare across forks" link to
+   enable the pull request to use your fork.  Set the "base repository" to
+   `IMAP-Science-Operations-Center` and "base" to `dev`.  Set the "head
+   repository" to the `imap_processing` repository under your personal fork
+   and "compare" to your feature branch. If the feature branch is still under
+   development, you can click the "Convert to draft" button under the
+   "Reviewers" section, or add a "[WIP]" at the beginning of the pull request
+   title to signify that the pull request is not ready to be merged.
+
+10. Assign at least one reviewer to the pull request.  They will review your
```

Comment:
Good idea. I'm personally in favor of 2+ reviewers on most occasions, especially since we have a relatively large development team.  Maybe we can say something like "Assign at least one reviewer, but 2 or more reviewers are highly encouraged"?

Reply:
Yeah, that sounds good